src/snowflake/connector/util_text.py

```diff
@@ -33,6 +43,109 @@ def split_statements(
     buf: StringIO,
     remove_comments: bool = False,
     delimiter: Optional[SQLDelimiter] = None,
+):
+    """Entry function to split statements."""
+    if delimiter is None or str(delimiter) == ";":
+        return split_statements_new(buf, remove_comments)
+    else:
+        return split_statements_old(buf, remove_comments, delimiter)
```

Comment:
Please don't call this `new` vs `old`. I'd much rather have them numbered.
Otherwise what are we going to call the next version, new new?